lib/pbench/server/cache_manager.py

```diff
@@ -464,13 +464,15 @@ def filestream(self, path: str) -> Optional[dict]:
         info = self.get_info(file_path)
         info["stream"] = None
 
-        if info["type"] == CacheType.FILE or file_path.name == self.name:
-            try:
+        try:
+            if not path:
+                info["stream"] = self.tarball_path.open("rb")
+            if info["type"] == CacheType.FILE:
```

Comment:
what is `info["type"]` for the tarball path? In any case, it'd be safer for this to be an `elif` so it's unambiguously clear that we're not going to overwrite `info["stream"]`!